Lesson21/nurbasak/script.js

```diff
@@ -0,0 +1,168 @@
+/*
+1. Declare variable that store elements that we need to use for this task
+
+2. Add event listeners (on click - buttons , on change - inputs)
+
+3. Take data from current element and show it in the target output field.
+
+*/
+
+const onlyLetters = /^[a-zA-Z\s]+$/;
+const emailPattern = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+
+const dateInput = document.getElementById('date');
+const dateOutput = document.getElementById('selected-date');
+const timeslotButtons = document.querySelectorAll('.slot');
+const selectedTimeOutput = document.getElementById('selected-time');
+const form = document.getElementById('booking_form');
+const confirmation = document.getElementById('confirmation');
+const confirmedDate = document.getElementById('confirmed-date');
+const confirmedTime = document.getElementById('confirmed-time');
+const confirmButton = document.getElementById('confirm');
+
+const nameInput = document.getElementById("name");
+const nameInputError = document.getElementById("nameError");
+const nameOutput = document.getElementById("given-username");
+const emailInput = document.getElementById("email");
+const emailInputError = document.getElementById("emailError");
+const emailOutput = document.getElementById("given-email");
+const confirmedName = document.getElementById('confirmed-name');
+const confirmedEmail = document.getElementById('confirmed-email');
+
+let formValid = true;
+
+// TODO: set min date for tomorrow or any other future date
+
+
+const data = {
+  date: null,
+  time: null,
+  name: null,
+  email:null,
+};
+
+//added personal info part to the form
+
+function validateName(name,nameError) {
+  if (!onlyLetters.test(name)) {
+    nameError.innerText = 'Please enter a valid username.';
+    nameError.classList.remove('hidden');
+    formValid = false;
+  } else if (!(name.trim().length > 0)) {
+    nameError.classList.remove('hidden');
+    nameError.innerText = 'Enter more than one character.';
+    formValid = false;
+  } else {
+    nameError.innerText = '';
+    nameError.classList.add('hidden');
+  }
+
+  nameInput.addEventListener('change', () => {
```

Comment:
Adding an event listener inside a function like this can lead to multiple listeners being attached, causing performance issues and unexpected behavior.
Move the change event listener registration outside the validation function, so it’s only added once.